src/commands/add/addCredential.ts

```diff
@@ -22,40 +18,31 @@ import { TargetScope } from '../../types/targetScope'
  * The file will contain the client ID, secret, access token and refresh token if the server type of target is viya.
  * The file will contain the username and password if the server type of target is sas9
  * Its name will be of the form `.env.{targetName}`
- * @param {string} targetName- the name of the target to create the env file for.
+ * @param {string} target- the target to create the env file for.
  * @param {boolean} insecure- boolean to use insecure connection, default is false. lf true the server will not reject any connection which is not authorized with the list of supplied CAs
  */
 export const addCredential = async (
-  targetName: string,
+  target: Target,
   insecure: boolean = false,
-  targetScope?: TargetScope
+  targetScope: TargetScope
 ): Promise<void> => {
-  targetName = validateTargetName(targetName)
   let scope = ''
-  let { target, isLocal } = await findTargetInConfiguration(
```

Comment:
All this is handled by the `TargetCommand` class and the `target` object is passed into this function directly..